src/Bicep.Cli.IntegrationTests/LintCommandTests.cs

```diff
@@ -208,6 +208,50 @@ public async Task Lint_with_warnings_should_log_warnings_and_return_0_exit_code(
         error.Should().StartWith($"{inputFile}(4,7) : Warning no-unused-params: Parameter \"notUsedParam\" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]");
     }
 
+    [TestMethod]
+    public async Task Lint_bicepparam_with_warnings_should_log_warnings_and_return_0_exit_code()
+    {
+        var outputPath = FileHelper.GetUniqueTestOutputPath(TestContext);
+        var bicepFile = FileHelper.SaveResultFile(TestContext, "main.bicep", @"
+@minValue(1)
+@maxValue(50)
+param notUsedParam int
+", outputPath);
+        var inputFile = FileHelper.SaveResultFile(TestContext, "main.bicepparam", @"
+using 'main.bicep'
+param notUsedParam = 3
+", outputPath);
+
+        var (output, error, result) = await Bicep("lint", inputFile);
+
+        result.Should().Be(0);
+        output.Should().BeEmpty();
+        error.Should().StartWith($"{bicepFile}(4,7) : Warning no-unused-params: Parameter \"notUsedParam\" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]");
+    }
+
+    [TestMethod]
+    public async Task Lint_bicepparam_with_error_should_fail()
+    {
+        var outputPath = FileHelper.GetUniqueTestOutputPath(TestContext);
+        var bicepFile = FileHelper.SaveResultFile(TestContext, "main.bicep", @"
+@minValue(1)
+@maxValue(50)
+param notUsedParam int
+", outputPath);
+        var inputFile = FileHelper.SaveResultFile(TestContext, "main.bicepparam", @"
+using 'main.bicep'
+param notUsedParm = 'string'
+", outputPath);
+
+        var (output, error, result) = await Bicep("lint", inputFile);
+
+        result.Should().Be(1);
+        output.Should().BeEmpty();
+        error.Should().Contain($"{bicepFile}(4,7) : Warning no-unused-params: Parameter \"notUsedParam\" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]");
```

Comment:
Should this have been emitted? It's a diagnostic about the template mentioned in the `using` statement rather than about the .bicepparam file.

(If linting a template with modules causes diagnostics from those modules to be surfaced, then this is probably the correct behavior.)

Reply:
Personally, I think it's reasonable to emit linter diagnostics for linked Bicep files, and it matches the behavior of `bicep build-params`.